sqlglot/dialects/tsql.py

```diff
@@ -281,8 +282,16 @@ class Tokenizer(tokens.Tokenizer):
         QUOTES = ["'", '"']
         HEX_STRINGS = [("0x", ""), ("0X", "")]
 
+        COMMAND_PREFIX_TOKENS = {TokenType.SEMICOLON}
+
         KEYWORDS = {
             **tokens.Tokenizer.KEYWORDS,
+            "BEGIN": TokenType.COMMAND,
+            "BEGIN TRANSACTION": TokenType.BEGIN,
+            "COMMIT": TokenType.COMMIT,
```

Comment:
these are unnecessary, why did you add them?

Reply:
"BEGIN" and "BEGIN TRANSACTION" mappings are required to disambiguate the flow control BEGIN..END and the transactional uses of BEGIN.

Reply:
COMMIT and ROLLBACK can be removed.

Reply:
Removed these tokens.